pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -1122,9 +1120,7 @@ func getTaskrunAnnotations(pr *v1.PipelineRun) map[string]string {
 	for key, val := range pr.ObjectMeta.Annotations {
 		annotations[key] = val
 	}
-	return kmap.Filter(annotations, func(s string) bool {
-		return filterReservedAnnotationRegexp.MatchString(s)
-	})
+	return kmap.Filter(annotations, pipeline.FilterReservedAnnotation)
```

Comment:
@vdemeester I meant why not just in webhook? I see that we also do it during propagation. This will anyway run in webhook, isn't it so?

Reply:
@khrm mainly because, for several reason, there is no real guarantee that it would have happened. For example, for any resolved object (using the deprecated bundle or the reslovers) didn't get through the mutation webhook — and thus still could have those annotations. There could also be a setup where the webhook is not there at all, and thus same thing.

If you look at the reconciler code, this is true for a lot of things, this is why the reconciler calls `.SetDefaults()` or `.Validate()` on objects.